src/core/namespace.c

```diff
@@ -168,16 +167,21 @@ static int append_mounts(BindMount **p, char **strv, MountMode mode) {
 
         STRV_FOREACH(i, strv) {
                 bool ignore = false;
+                char *path;
 
                 if (IN_SET(mode, INACCESSIBLE, READONLY, READWRITE) && startswith(*i, "-")) {
                         (*i)++;
                         ignore = true;
                 }
 
-                if (!path_is_absolute(*i))
+                path = strdup(*i);
+                if (!path)
+                        return -ENOMEM;
+
+                if (!path_is_absolute(path))
```

Comment:
It seems that `strdup` should be done below the `!path_is_absolute` check. This would avoid the memleak.